cmd/common.go

```diff
@@ -614,3 +631,116 @@ func exportCodesignFiles(tool Tool, archivePath, outputDirPath string) error {
 
 	return nil
 }
+
+func askUploadProvProfiles() (bool, error) {
+	messageToAsk := "Do you want to upload the provisioning profiles to Bitrise?"
+
+	answer, err := goinp.AskForBoolFromReader(messageToAsk, os.Stdin)
+	if err != nil {
+		return answer, err
+	}
+
+	return answer, nil
+}
+
+// ----------------------------------------------------------------
+// --- Upload methods
+
+// TODO
+func uploadProvisioningProfiles(profilesToUpload []profileutil.ProvisioningProfileInfoModel, outputDirPath string) error {
+	// Get accesToken from stdin
+	accessToken, err := uploaders.AskAccessToken()
+
+	appList, err := uploaders.FetchMyApps(accessToken)
+	if err != nil {
+		log.Errorf("Failed to ask your app list from Bitrise: %s", err)
+		return err
+	}
+
+	selectedApp, err := selectApp(appList)
+	if err != nil {
+		return err
+	}
+
+	// Upload provisioning profiles
+	for _, profile := range profilesToUpload {
+
+		exportFileName := provProfileExportFileName(profile, outputDirPath)
+
+		provProfile, err := os.Open(outputDirPath + "/" + exportFileName)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := provProfile.Close(); err != nil {
+				log.Warnf("Provisioning profile close failed, err: %s", err)
+			}
+
+		}()
+
+		info, err := provProfile.Stat()
+		if err != nil {
+			return err
+		}
+
+		bytes := info.Size()
+		log.Debugf("\n%s size: %d", exportFileName, bytes)
+
+		provProfSlugResponseData, err := uploaders.RegisterProvisioningProfile(accessToken, selectedApp.Slug, bytes, profile)
+		if err != nil {
+			return err
+		}
+
+		err = uploaders.UploadProvisioningProfile(provProfSlugResponseData, outputDirPath, exportFileName)
+		if err != nil {
```

Comment:
if function returns only a single error, define and check the error in if statement:

if err := uploaders.UploadProvisioningProfile(provProfSlugResponseData, outputDirPath, exportFileName); err != nil {
}

avoid using `err = xyz()` this can easily lead to a shadowed error! always try to define the error like: `err := xyz()`